src/Network/NWEthernetChannel.cs

```diff
@@ -0,0 +1,208 @@
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_ethernet_channel nw_ethernet_channel_create (ushort ether_type, OS_nw_interface networkInterface);
+
+		// we cannot pass an enum! As per docs in the headers:
+		// The custom EtherType to be used for all Ethernet frames in this channel. The
+		// EtherType is the two-octet field in an Ethernet frame, indicating the protocol
+		// encapsulated in the payload of the frame.  This parameter is in little-endian
+		// byte order.  Only custom EtherType values are supported. This parameter cannot
+		// be an EtherType already handled by the system, such as IPv4, IPv6, ARP, VLAN Tag,
+		// or 802.1x.
+		// 
+		// Calling processes must hold the "com.apple.developer.networking.custom-protocol"
+		// entitlement.
+		public NWEthernetChannel (ushort ethernetType, NWInterface networkInterface)
+		{
+			if (networkInterface == null)
+				throw new ArgumentNullException (nameof (networkInterface));
+
+			InitializeHandle (nw_ethernet_channel_create (ethernetType, networkInterface.Handle));
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_ethernet_channel_start (OS_nw_ethernet_channel ethernet_channel);
+
+		public void Start () => nw_ethernet_channel_start (GetCheckedHandle ()); 
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_ethernet_channel_cancel (OS_nw_ethernet_channel ethernet_channel);
+
+		public void Cancel () => nw_ethernet_channel_cancel (GetCheckedHandle ());
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_ethernet_channel_set_queue (OS_nw_ethernet_channel ethernet_channel, IntPtr queue);
+
+		public void SetQueue (DispatchQueue queue)
+		{
+			if (queue == null)
+				throw new ArgumentNullException (nameof (queue));
+			nw_ethernet_channel_set_queue (GetCheckedHandle (), queue.Handle);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_ethernet_channel_send (OS_nw_ethernet_channel ethernet_channel, OS_dispatch_data content, ushort vlan_tag, string remote_address, ref BlockLiteral completion);
+
+		delegate void nw_ethernet_channel_send_completion_t (IntPtr block, IntPtr error);
+		static nw_ethernet_channel_send_completion_t static_SendCompletion = TrampolineSendCompletion;
+
+		[MonoPInvokeCallback (typeof (nw_ethernet_channel_send_completion_t))]
+		static void TrampolineSendCompletion (IntPtr block, IntPtr error)
+		{
+			var del = BlockLiteral.GetTarget<Action<NWError>> (block);
+			if (del != null) {
+				var err = error == IntPtr.Zero ? null : new NWError (error, owns: false);
+				del (err);
+				err?.Dispose ();
+			}
+		}
+
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public void Send (ReadOnlySpan<byte> content, ushort vlanTag, string remoteAddress, Action<NWError> callback)
+		{
+			if (callback == null)
+				throw new ArgumentNullException (nameof (callback));
+
+			using (var dispatchData = DispatchData.FromByteBuffer (content.ToArray (), 0, content.Length)) {
```

Comment:
DispatchData should have a ROSpan overload

Reply:
worth filing an issue (not the right PR to do this) so we don't forget

Reply:
@spouliot @Therzok take a look https://github.com/xamarin/xamarin-macios/issues/7446 any other possible methods to add?